straxen/plugins/event_shadow.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import strax
+import numba
+export, __all__ = strax.exporter()
+
+@export
+@strax.takes_config(
+    strax.Option('pre_s2_area_threshold', default=1000,
+                 help='Only take S2s large than this into account when calculating EventShadow.'),
+    strax.Option('time_window_backward', default=int(3e9),
+                 help='Search for S2s causing shadow in this time window'))
+class EventShadow(strax.Plugin):
+    """
+    This plugin can find and calculate the previous S2 shadow at event level,
+    with time window backward and previous S2 area as options.
+    It also gives the area and position infomation of these previous S2s.
+    """
+    depends_on = ('event_basics','peak_basics','peak_positions')
+    parallel = False
+    provides = 'event_shadow'
+    __version__ = '0.0.6'
+
```

Comment:
One question. I guess this plugin will mainly be needed to optimize the corresponding cut is not it? Since the plugin is now reasonably fast I would suggest that we make it a `save_when = strax.SaveWhen.EXPLICIT`-plugin. In that case the output is only stored if asked explicitly.